core/templates/dev/head/services/speech-synthesis-chunker.service.spec.ts

```diff
@@ -32,87 +32,212 @@ describe('Speech Synthesis Chunker Service', () => {
+      var latex1 = '\\\\sin{90}';
+      var latex2 = '\\\\cos{0}';
+      var latex3 = '\\\\tan{uv}';
       var speakableLatex1 =
         speechSynthesisChunkerService.formatLatexToSpeakableText(latex1);
       var speakableLatex2 =
         speechSynthesisChunkerService.formatLatexToSpeakableText(latex2);
       var speakableLatex3 =
         speechSynthesisChunkerService.formatLatexToSpeakableText(latex3);
-      var speakableLatex4 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex4);
-      expect(speakableLatex1).toEqual('x^2');
-      expect(speakableLatex2).toEqual('42 to the power of 4');
-      expect(speakableLatex3).toEqual('x to the power of 62');
-      expect(speakableLatex4).toEqual('3n to the power of 4x');
-    }
-  );
-
-  it ('Should properly convert trigonometric functions in LaTeX to ' +
-      'speakable text', () => {
-    var latex1 = '\\\\sin{90}';
-    var latex2 = '\\\\cos{0}';
-    var latex3 = '\\\\tan{uv}';
-    var speakableLatex1 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex1);
-    var speakableLatex2 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex2);
-    var speakableLatex3 =
-        speechSynthesisChunkerService.formatLatexToSpeakableText(latex3);
-    expect(speakableLatex1).toEqual('the sine of 90');
-    expect(speakableLatex2).toEqual('the cosine of 0');
-    expect(speakableLatex3).toEqual('the tangent of u v');
+      expect(speakableLatex1).toEqual('the sine of 90');
+      expect(speakableLatex2).toEqual('the cosine of 0');
+      expect(speakableLatex3).toEqual('the tangent of u v');
+    });
+  });
+
+  describe('convertToSpeakableText', () => {
+    it('should properly convert the raw_latex-with-value attribute to' +
+      ' speakable text', () => {
+      const html = (
+        '<oppia-noninteractive-math raw_latex-with-value="5 - 1">' +
+        '</oppia-noninteractive-math>' +
+        '<li>Speech</li>' +
+        '<li>Text</li>'
+      );
+
+      expect(speechSynthesisChunkerService.convertToSpeakableText(html))
+        .toBe('5 minus 1 Speech. Text. ');
+    });
+
+    it('should properly convert the text-with-value attribute to' +
+      ' speakable text', () => {
+      const html = (
+        '<oppia-noninteractive-link text-with-value="&quot;">' +
+        '</oppia-noninteractive-link>' +
+        '<li>"Speech"</li>' +
+        '<li>Text</li>'
+      );
+
+      expect(speechSynthesisChunkerService.convertToSpeakableText(html))
+        .toBe('Speech. Text. ');
+    });
+  });
+
+  describe('speak', function() {
+    const MockSpeechSynthesisUtteranceConstructor = (
+      SpeechSynthesisUtterance);
+    const mockSpeechSynthesisUtteran = {
+      speak: () => {},
+      onend: () => {}
+    };
+
+    beforeEach(() => {
+      spyOn(window, 'SpeechSynthesisUtterance').and.returnValue(
+        // @ts-ignore mock doesn't have all property and methods of a native
+        // SpeechSynthesisUtterance
```

Comment:
Add a period at the end. 

/cc @ankita240796, do we think it's fine to leave this as-is? or we can do something over here?

Reply:
Done.